src/node_options.cc

```diff
@@ -86,6 +86,11 @@ void PerIsolateOptions::CheckOptions(std::vector<std::string>* errors) {
     return;
   }
 
+  if (per_process::cli_options->report_on_fatalerror) {
+    errors->push_back("--report-on-fatalerror option is valid only when "
+                      "--experimental-report is set");
+  }
+
```

Comment:
This is one extra change I am doing than https://github.com/nodejs/node/pull/29881, to fix the issue mentioned in https://github.com/nodejs/node/pull/29881#issuecomment-581345483